modules/Core/widgets/ReactionsProfileWidget.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+class ReactionsProfileWidget extends ProfileWidgetBase {
+
+    private static array $_collectors = [
+        'recieved' => [],
+        'given' => []
+    ];
+
+    public function __construct(Smarty $smarty) {
+        parent::__construct($smarty);
+
+        $widget_query = self::getData('Reactions');
+
+        $this->_name = 'Reactions';
+        $this->_order = $widget_query->order;
+        $this->_description = "Displays a user\'s recieved reactions on their profile.";
+        $this->_module = 'Core';
+        $this->_location = $widget_query->location;
+    }
+
+    public function initialise(User $user): void {
+        $reactions = [];
+        foreach (Reaction::find(true, 'enabled') as $reaction) {
+            $reactions[$reaction->id] = [
+                'name' => $reaction->name,
+                'html' => Text::renderEmojis($reaction->html),
+                'recieved' => 0,
+                'given' => 0,
+            ];
+        }
+        foreach (self::$_collectors['recieved'] as $collector) {
+            $received = $collector($user);
+            foreach ($received as $reaction) {
+                $reactions[$reaction->reaction_id]['recieved']++;
+            }
+        }
+        foreach (self::$_collectors['given'] as $collector) {
+            $given = $collector($user);
+            foreach ($given as $reaction) {
+                $reactions[$reaction->reaction_id]['given']++;
+            }
+        }
+        // Sort by most recieved
+        usort($reactions, static function ($a, $b) {
+            return $b['recieved'] <=> $a['recieved'];
+        });
+        $this->_smarty->assign([
+            'ALL_REACTIONS' => $reactions,
+        ]);
+        $this->_content = $this->_smarty->fetch('widgets/forum/reactions.tpl');
+    }
+
+    public static function addRecievedCollector(Closure $collector): void {
```

Comment:
"Collectors" (open to different word), let modules define ways to calculate reactions a user has given/received.

This means the widget does not rely on the forum module specifically, and will work without it.
It also opens doors for other modules to display reaction counts here - maybe if their Resource got a like, or Suggestion got a like/dislike! cc @partydragen

Reply:
This will be updated to a system called `reaction contexts`.
It will allow modules to hook into our reactions queries file so they don't need to repeat the same logic for their reaction implementations (suggestions, resources module)

A `ReactionContext` class will likely contain the following methods:
- `name`
- `getUserReceived(User $user)`
- `getUserGiven(User $user)`
- `validateReactable(string $id)` ensure that this "reactable" object exists
- `hasReacted(User $user, Reaction $reaction, string $reactable_id)` - `reactable_id` being the ID of a profile post/forum post
- `giveReaction(User $from, User $reciever, Reaction $reaction, string $reactable_id)`
- `deleteReaction(User $from, User $reciever, Reaction $reaction, string $reactable_id)` - nameless will handle firing the events, these methods are for the reactioncontext to update its DB records and do any other tasks (logging)